File: src/main/java/com/just/donate/flow/Expense.java
package com.just.donate.flow;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import java.util.UUID;

public class Expense {
    private final UUID id;
    private final Date date;
    private final BigDecimal amount;
    private List<DonationPart> donationParts;

    public Expense(Date date, BigDecimal amount) {
        this.id = UUID.randomUUID();
        this.date = date;
        this.amount = amount;
        this.donationParts = new ArrayList<>();
    }

    public BigDecimal getAmount() {
        return amount;
    }

    public void addDonationPart(DonationPart donationPart) {
        this.donationParts.add(donationPart);
    }
}

File: backend/src/main/scala/com/just/donate/models/Expense.scala
package com.just.donate.models

import com.just.donate.utils.Money

import java.time.LocalDateTime
import java.util.UUID

case class Expense(
  description: String,
  amount: Money,
  earMarking: Option[Earmarking] = None,
  paidBy: Seq[DonationPart] = Seq.empty,
  fromAccount: String,
  time: LocalDateTime = LocalDateTime.now,
  id: String = UUID.randomUUID().toString
):

  def isBound: Boolean = earMarking.isDefined
